src/backend/cpu/ireduce.cpp

```diff
@@ -63,6 +63,7 @@ T ireduce_all(unsigned *loc, const Array<T> &in) {
     af::dim4 dims    = in.dims();
     af::dim4 strides = in.strides();
     const T *inPtr   = in.get();
+    dim_t idx = 0;
 
```

Comment:
The  getQueue().sync(); of line 62 has to move to line 66.

For incomplete JIT arrays, the evaluation is initiated on line 65 in.get().  The sync has to be afterwards to have this completed before the following loop.

Test case:
array jit(dim4(2),{1.0f, 2.0f};
jit += constant(1.0f,dim4(2));
float val; unsigned idx;
min(&val, &idx, jit);

The min function will be calculated on the original content, without having added the 1.0;